app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchTEPresenter.java

```diff
@@ -810,6 +810,14 @@ public int getEnrollmentColor() {
             return -1;
     }
 
+    @Override
+    public boolean programHasAssignment(String programUid) {
+        return !d2.programModule().programStages()
+                .byProgramUid().eq(programUid)
+                .byEnableUserAssignment().isTrue()
+                .blockingIsEmpty();
+    }
```

Comment:
can we check here if its true or false and then do 

view.addAsignedToMe

or view.removeAssignedToMe

For the future we could unit test this logic

Reply:
and we remove logic and responsability from the view. Since it's the presenter the one to decide to show or hide... ?